### DISCUSSION TITLE
Boilerplate, oque não pode faltar?

### DISCUSSION THREAD

**Discussion by Juuhz** [frontendbr/forum#1787] [Discussão]
Fala pessoal, na paz dos códigos?

Gostaria de começar uma discussão sobre `boilerplates`, estou querendo criar um na empresa que trabalho, para facilitar a fase inicial do projeto. 

esse boilerplate, servirá para projetos relacionados ao front (Landing Pages / Hotsites).

1) Vocês usam na empresa de vocês algo parecido?
2) Quais metodologias/libs estão usando?

OBS: Meu foco é o React (NextJs).

Abraços.

**Comment by didiraja** [comment]
Nesse formato de Hotsite e landing page, eu acho que o básico resolve: Webpack, Babel, React com SSR, lib de CSS a sua escolha e uns lints de ES e Prettier pra manter o código organizado.

**Comment by NegriHenrique** [comment]
Cara eu gosto de usar TypeScript com o Next e um sass, e para ficar ainda mais organizado o código eu sempre uso o eslint com prettier

**Comment by dionarodrigues** [comment]
@Juuhz Os requisitos do boilerplate irão depender muito do objetivo final:

- quais tecnologias a empresa utiliza para desenvolver o site?
- como se dá o processo de criação do HTML, CSS e JS?
- como geralmente vocês lidam com as otimizações de imagens, fontes...?

Uma vez tendo todo esse processo claro fica mais fácil você buscar exemplos de boilerplate que atendem ãs suas necessidades e, assim, buscar inspiração.

Eu, por exemplo criei esse [boilerplate com Gulp, Pug, Stylus e demais configurações](https://github.com/diogorodrigues/iceberg-boilerplate) em uma época em que eu costumava usar bastante essas tecnologias.

Recentemente eu crei esse [starter multilíngue para Gatsby](https://github.com/diogorodrigues/iceberg-gatsby-multilang) já que venho desenvolvendo alguns projetos com essa tecnologia. 

Mas também criei alguns starters para projetos em [ReactJS](https://github.com/diogorodrigues/reactjs-typescript-starter), [React Native](https://github.com/diogorodrigues/react-native-typescript-starter) e [NodeJS](https://github.com/diogorodrigues/node-typescript-starter) com configurações de typescript, lint, styled components... que são tecnologias que venho utilizando com frequência. 

Então, a minha dica é listar o que você precisa primeiro e depois buscar soluções parecidas, experimentar elas e depois desenvolver a sua própria, caso seja necessário.

**Comment by tarciozemel** [comment]
Dica adicional: atenha-se ao "MMC" dos projetos aí da empresa, já que se trata de um ponto de partida.

Não adianta fazer um boilerplate em que seja possível fazer 1001 coisas se, no final do dia, só se fazem 20.
